src/logger.ts

```diff
@@ -128,14 +128,26 @@ interface PrefixedLogger extends loglevel.Logger, LoggerWithLogMethod {
  *
  * @param prefix Prefix to add to each logged line. If undefined, no prefix will be added.
  */
-function getPrefixedLogger(prefix?: string): LoggerWithLogMethod {
+function getPrefixedLogger(prefix?: string): PrefixedLogger {
     const loggerName = DEFAULT_NAMESPACE + (prefix === undefined ? "" : `-${prefix}`);
     const prefixLogger = loglevel.getLogger(loggerName) as PrefixedLogger;
-
+    // This is a save cast since export const logger is constructed with getPrefixedLogger
+    // It is downcastet to `Logger` to minimize the public/exported api.
     if (prefixLogger.getChild === undefined) {
         // This is a new loglevel Logger which has not been turned into a PrefixedLogger yet.
         prefixLogger.prefix = prefix;
-        prefixLogger.getChild = (childPrefix): Logger => getPrefixedLogger((prefix ?? "") + childPrefix);
+        prefixLogger.getChild = (childPrefix): Logger => {
+            const rootLogger = logger as PrefixedLogger;
+            // create the new child logger
+            const childLogger = getPrefixedLogger((prefix ?? "") + childPrefix);
+            // assign the same methodFactory as the root logger.
+            // this is useful if we add extensions to the root logger that modify
+            // its methodFactory. (an example extension is: storing each log to a rageshake db)
+            childLogger.methodFactory = rootLogger.methodFactory;
```

Comment:
I'm wondering if this should instead be

```suggestion
            childLogger.methodFactory = prefixLogger.methodFactory;
```

so that if I extend a *child* of the root logger, then any *children of the child* will inherit their parent's extensions, rather than just the root extension

Reply:
Yes this makes sense.
It would still give all children the same factory in the EC usecase sine we extend the root logger and all children will rebuild with their parent so it will inherit downwards.

Maybe its also important to mention that getChild and adding extensions is order dependent. Since we will rebuild only at the time where we add a child later changes to the factory will not apply even though the factory is shared between all loggers.
Because in both ways it is different if we first create all the children and then extend the root logger or if we extend the root logger and than create the children.

Reply:
Sure, wouldn't hurt to mention

Reply:
added this comment to the `getChild` docs